File: py/packages/over-redis/tests/multi_service_example.py
from clamator_over_redis import RedisRpcClient

from .generated.arith import AddParams, AddResult, ArithClient
from .generated.logger import LoggerClient, LogParams


# One key_prefix-pinned RedisRpcClient backs many service proxies.
async def call_multiple_services(key_prefix: str) -> AddResult:
    client = RedisRpcClient(key_prefix=key_prefix)
    await client.start()
    arith = ArithClient(client)
    logger = LoggerClient(client)
    r = await arith.add(AddParams(a=2, b=3))
    await logger.log(LogParams(msg=f"sum={r.sum}"))
    await client.stop()
    return r

File: ts/packages/over-redis/tests/multi-service.example.ts
import { RedisRpcClient } from '../src/index.js';
import { ArithClient } from './generated/arith.js';
import { LoggerClient } from './generated/logger.js';

// One keyPrefix-pinned RedisRpcClient backs many service proxies.
export async function callMultipleServices(keyPrefix: string) {
  const client = new RedisRpcClient({ keyPrefix });
  await client.start();
  const arith = new ArithClient(client);
  const logger = new LoggerClient(client);
  const sum = await arith.add({ a: 2, b: 3 });
  await logger.log({ msg: `sum=${sum.sum}` });
  await client.stop();
  return sum;
}
